From daa34105d96d2b8eda9526a237b4d4721f7fc7a8 Mon Sep 17 00:00:00 2001
From: Konstantin V Shvachko <shv@apache.org>
Date: Tue, 11 Jul 2017 16:01:58 -0700
Subject: [PATCH 2596/2848] HADOOP-14586. StringIndexOutOfBoundsException
 breaks org.apache.hadoop.util.Shell on 2.7.x with
 Java 9. Contributed by Uwe Schindler.

Change-Id: I4d72e2893dd75c70c5a7273b22345e837a9b356f
---
 .../main/java/org/apache/hadoop/util/Shell.java    |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java
index 87320f9..4746732 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java
@@ -47,8 +47,11 @@
   
   public static final Log LOG = LogFactory.getLog(Shell.class);
   
-  private static boolean IS_JAVA7_OR_ABOVE =
-      System.getProperty("java.version").substring(0, 3).compareTo("1.7") >= 0;
+  /**
+   * java.version is enforced via maven-enforcer-plugin.
+   * Starting from Hadoop 2.7.0 Java 1.7 or higher is required.
+   */
+  private static boolean IS_JAVA7_OR_ABOVE = true;
 
   public static boolean isJava7OrAbove() {
     return IS_JAVA7_OR_ABOVE;
-- 
1.7.9.5

